### PR TITLE
[+] Adds pending approval status/message for private groups

### PR DESCRIPTION
**Pull request by andy-smith-cds**
[*] Allows users to join group again after being rejected

### CODE REVIEW

**Review by Matt-Beldon-CDS — APPROVED** [review]

File: futurenhs.app/components/GroupPageHeader/index.tsx [hunk 1]
```diff
@@ -21,6 +21,7 @@ import { useMediaQuery } from '@hooks/useMediaQuery'
 import { Theme } from '@appTypes/theme'
 
 import { Props } from './interfaces'
+import { status } from '@constants/status'
```
Comment (ReallySmall) [inline futurenhs.app/components/GroupPageHeader/index.tsx:24]: Can this name be made more specific to its purpose?

File: futurenhs.app/components/GroupPageHeader/index.tsx [hunk 2]
```diff
@@ -219,6 +222,17 @@ export const GroupPageHeader: (props: Props) => JSX.Element = ({
...
                             className={generatedClasses.actionsWrapper}
                         >
+                            {shouldRenderPendingMessage &&
+                                !shouldRenderGroupJoinLink && (
+                                    <div className="nhsuk-body-m">
+                                        <SVGIcon
+                                            name="icon-clock"
+                                            className="u-w-4 u-h-4 u-mr-2"
+                                        />
+                                        Awaiting Approval
```
Comment (ReallySmall) [inline futurenhs.app/components/GroupPageHeader/index.tsx:232]: Pass in via content config if possible

**Review by ReallySmall — CHANGES_REQUESTED** [review]

File: futurenhs.app/components/GroupPageHeader/index.tsx (hunk 1 above)
Comment (andy-smith-cds) [reply futurenhs.app/components/GroupPageHeader/index.tsx]: renamed to groupMemberStatus

File: futurenhs.app/components/GroupPageHeader/index.tsx (hunk 2 above)
Comment (andy-smith-cds) [reply futurenhs.app/components/GroupPageHeader/index.tsx:233]: Can't in this one unfortunately as it's not a page template so doesn't use the text service. Rest of the strings in this component are hardcoded - actions, join group etc.

**Review by ReallySmall — APPROVED** [review]

**Pull request merged**